skyportal/handlers/api/comment.py

```diff
@@ -320,17 +355,23 @@ def post(self, associated_resource_type, resource_id):
             for user_mentioned in users_mentioned_in_comment:
                 self.flow.push(user_mentioned.id, "skyportal/FETCH_NOTIFICATIONS", {})
 
-        if comment.obj.id:  # comment on object, or object related resources
+        if isinstance(comment, CommentOnGCN):
             self.push_all(
-                action='skyportal/REFRESH_SOURCE',
-                payload={'obj_key': comment.obj.internal_key},
+                action='skyportal/REFRESH_GCNEVENT',
+                payload={'gcnEvent_dateobs': comment.gcn.dateobs},
             )
+        else:
+            if comment.obj.id:  # comment on object, or object related resources
+                self.push_all(
+                    action='skyportal/REFRESH_SOURCE',
+                    payload={'obj_key': comment.obj.internal_key},
+                )
 
-        if isinstance(comment, CommentOnSpectrum):
-            self.push_all(
-                action='skyportal/REFRESH_SOURCE_SPECTRA',
-                payload={'obj_internal_key': comment.obj.internal_key},
-            )
+            if isinstance(comment, CommentOnSpectrum):
+                self.push_all(
+                    action='skyportal/REFRESH_SOURCE_SPECTRA',
+                    payload={'obj_internal_key': comment.obj.internal_key},
+                )
```

Comment:
Why do we need this in a separate else clause? If it is an instance of `CommentOnGCN` then it would not have an `obj`, right? Maybe we should just change `if comment.obj.id` to `if hasattr(comment, 'obj')`. In this case the GCN comment `if` can come after the object and spectrum comments. (do this also for PUT and DELETE).

Reply:
yeah you are right, fixing this.